src/components/tasks/change-proposals-drawer.tsx

```diff
@@ -54,6 +55,14 @@ const ChangeProposalsDrawer = ({
     [changeProposals],
   );
 
+  // Clears selected change proposal id when drawer is closed
+  useEffect(() => {
```

Comment:
It would be easier to handle this in click event handler, like

```
const onToggleOpen = () => {
  if (open) setSelectedChangeProposalId("");
  setOpen(!open);
}
```